src/diagnostics/core_diagnostics.jl

```diff
@@ -273,10 +273,38 @@ add_diagnostic_variable!(
     or from mixing length closure with EDMF SGS model.
     """,
     compute! = (out, state, cache, time) -> begin
+        turbconv_model = cache.atmos.turbconv_model
+
+        if isa(turbconv_model, PrognosticEDMFX) ||
+           isa(turbconv_model, DiagnosticEDMFX) ||
+           isa(turbconv_model, EDOnlyEDMFX)
+            ᶜmixing_length = mixing_length(state, cache)
+        else
```

Comment:
It's a little strange to have this calculation of the Smagorinsky-Lilly length here in the diagnostics, rather than where the mixing length is used dynamically. Wouldn't it be preferable for `mixing_length` to return Smag-Lilly when there is no TKE equation?

Maybe this doesn't need to be addressed right away, but if you agree it should be changed in the medium term, leaving a TODO and an explanation of why it is here would help.

Or could you just use `compute_gm_mixing_length` below? I know you are using the cached gradients here, but we want to remove them from the cache anyway.